core/src/main/scala/kafka/server/BrokerServer.scala

```diff
@@ -404,10 +404,6 @@ class BrokerServer(
         config.numIoThreads, s"${SocketServer.DataPlaneMetricPrefix}RequestHandlerAvgIdlePercent",
         SocketServer.DataPlaneThreadPrefix)
 
-      if (socketServer.controlPlaneRequestChannelOpt.isDefined) {
-        throw new RuntimeException(KafkaConfig.ControlPlaneListenerNameProp + " is not " +
-          "supported when in KRaft mode.")
-      }
```

Comment:
Removing since it is redundant -- we are now checking this when creating a KafkaConfig